hivemind/p2p/p2p_daemon.py

```diff
@@ -140,6 +140,11 @@ async def create(
         socket_uid = secrets.token_urlsafe(8)
         self._daemon_listen_maddr = Multiaddr(cls._UNIX_SOCKET_PREFIX + f"p2pd-{socket_uid}.sock")
         self._client_listen_maddr = Multiaddr(cls._UNIX_SOCKET_PREFIX + f"p2pclient-{socket_uid}.sock")
+        if announce_maddrs is not None:
+            for addr in announce_maddrs:
+                if addr.startswith("/ip4/") or addr.startswith("/ip6/"):
+                    if addr.endswith("/tcp/0") or addr.endswith("/udp/0/quic"):
+                        raise ValueError("Please specify an explicit port in announce_maddrs: port 0 is not supported")
```

Comment:
```suggestion
                addr = Multiaddr(addr)
                if ("tcp" in addr and addr["tcp"] == "0") or ("udp" in addr and addr["udp"] == "0"):
                    raise ValueError("Please specify an explicit port in announce_maddrs: port 0 is not supported")
```

This is a better way to work with multiaddrs. It accounts for some tricky cases where TCP/UDP is not in the end of the addr/doesn't use QUIC/etc.